packages/wallets/src/metamask/pages/walletOperations.page.ts

```diff
@@ -0,0 +1,104 @@
+import { Locator, Page, test } from '@playwright/test';
+
+export class WalletOperationPage {
+  page: Page;
+  nextButton: Locator;
+  confirmSignButton: Locator;
+  rejectButton: Locator;
+  cancelButton: Locator;
+  addTokenButton: Locator;
+  editGasFeeButton: Locator;
+  setHighGasFeeButton: Locator;
+  scrollRequestSignatureBlockButton: Locator;
+  rejectAllTxsButton: Locator;
+  confirmRejectAllTxsButton: Locator;
+  recipientButton: Locator;
+  recipientAddress: Locator;
+  popoverCloseButton: Locator;
+
+  constructor(page: Page) {
+    this.page = page;
+    this.nextButton = this.page.getByTestId('page-container-footer-next');
+    this.confirmSignButton = this.page.getByTestId('confirm-footer-button');
+    this.rejectButton = this.page.getByTestId('page-container-footer-cancel');
+    this.cancelButton = this.page.getByTestId('confirm-footer-cancel-button');
+    this.addTokenButton = this.page.locator('button:has-text("Add token")');
+    this.editGasFeeButton = this.page.getByTestId('edit-gas-fee-icon');
+    this.setHighGasFeeButton = this.page.getByTestId('edit-gas-fee-item-high');
+    this.scrollRequestSignatureBlockButton = this.page.getByTestId(
+      'signature-request-scroll-button',
+    );
+    this.rejectAllTxsButton = this.page.locator(
+      'div[class="page-container__footer-secondary"]',
+    );
+    this.confirmRejectAllTxsButton = this.page.locator(
+      'button:has-text("Reject all")',
+    );
+    this.recipientButton = this.page.locator(
+      '.sender-to-recipient__party--recipient-with-address',
+    );
+    this.recipientAddress = this.page.locator('input[id="address"]');
+    this.popoverCloseButton = this.page.locator('button[aria-label="Close"]');
+  }
+
+  async rejectAllTxInQueue() {
+    //Is there is any tx in queue.
+    try {
+      await this.rejectButton.waitFor({
+        state: 'visible',
+        timeout: 1000,
+      });
+    } catch (er) {
+      return;
+    }
+
+    if (await this.rejectAllTxsButton.isVisible()) {
+      await this.rejectAllTxsButton.click();
+      await this.confirmRejectAllTxsButton.click();
+    } else {
+      await this.rejectButton.click();
+    }
+  }
+
+  async rejectTransaction() {
+    await this.rejectButton.click();
+  }
+
+  async cancelSignTransaction() {
+    await this.cancelButton.click();
+  }
+
+  async signTransaction() {
+    await this.confirmSignButton.click();
+  }
+
+  async confirmTransactionOfTokenApproval() {
```

Comment:
could you add test.step for this method? and for confirmTransaction and assertReceiptAddress?

Reply:
we have step in the parent function in the `MetamaskApp.ts` file around `confirmTransaction` and `assertReceiptAddress` functions